plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/bitbucketServerPullRequest.ts

```diff
@@ -0,0 +1,266 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { InputError } from '@backstage/errors';
+import {
+  getBitbucketServerRequestOptions,
+  ScmIntegrationRegistry,
+} from '@backstage/integration';
+import { createTemplateAction } from '@backstage/plugin-scaffolder-node';
+import fetch, { RequestInit, Response } from 'node-fetch';
+import { parseRepoUrl } from './util';
+import { Config } from '@backstage/config';
+
+const createPullRequest = async (opts: {
+  project: string;
+  repo: string;
+  title: string;
+  description?: string;
+  toRef: object;
+  fromRef: object;
+  authorization: string;
+  apiBaseUrl: string;
+}) => {
+  const {
+    project,
+    repo,
+    title,
+    description,
+    toRef,
+    fromRef,
+    authorization,
+    apiBaseUrl,
+  } = opts;
+
+  let response: Response;
+  const data: RequestInit = {
+    method: 'POST',
+    body: JSON.stringify({
+      title: title,
+      description: description,
+      state: 'OPEN',
+      open: true,
+      closed: false,
+      locked: true,
+      toRef: toRef,
+      fromRef: fromRef,
+    }),
+    headers: {
+      Authorization: authorization,
+      'Content-Type': 'application/json',
+    },
+  };
+
+  try {
+    response = await fetch(
+      `${apiBaseUrl}/projects/${project}/repos/${repo}/pull-requests`,
+      data,
+    );
+  } catch (e) {
+    throw new Error(`Unable to create pull-reqeusts, ${e}`);
+  }
+
+  if (response.status !== 201) {
+    throw new Error(
+      `Unable to create pull requests, ${response.status} ${
+        response.statusText
+      }, ${await response.text()}`,
+    );
+  }
+
+  const r = await response.json();
+  return `${r.links.self[0].href}`;
+};
+const findBranches = async (opts: {
+  project: string;
+  repo: string;
+  branchName: string;
+  authorization: string;
+  apiBaseUrl: string;
+}) => {
+  const { project, repo, branchName, authorization, apiBaseUrl } = opts;
+
+  let response: Response;
+  const options: RequestInit = {
+    method: 'GET',
+    headers: {
+      Authorization: authorization,
+      'Content-Type': 'application/json',
+    },
+  };
+
+  try {
+    response = await fetch(
+      `${apiBaseUrl}/projects/${project}/repos/${repo}/branches?boostMatches=true&filterText=${branchName}`,
+      options,
+    );
+  } catch (e) {
+    throw new Error(`Unable to get branches, ${e}`);
+  }
+
+  if (response.status !== 200) {
+    throw new Error(
+      `Unable to get branches, ${response.status} ${
+        response.statusText
+      }, ${await response.text()}`,
+    );
+  }
+
+  const r = await response.json();
+  let branch;
+  for (const object of r.values) {
+    if (object.displayId === branchName) {
+      branch = object;
```

Comment:
could just return here, and return undefined at the bottom (personally not fond of declaring empty vars, but it's a matter of taste and i know some people absolutely abhor multiple returns, so feel free to ignore)

Reply:
changed it.